core/base/mtx_io.cpp

```diff
@@ -1005,4 +1009,43 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(GKO_DECLARE_WRITE_BINARY_RAW);
 GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(GKO_DECLARE_READ_GENERIC_RAW);
 
 
+void write(std::ostream& os, ptr_param<const LinOp> matrix)
+{
+    auto op = matrix.get();
+    run<
+#if GINKGO_ENABLE_HALF
+        gko::WritableToMatrixData<gko::float16, int32>,
```

Comment:
This makes me wonder whether we should change the `GKO_INSTANTIATE_FOR_...` macros into true X macros without the `template` prefix and `;` suffix